authz/sdk_end2end_test.go

```diff
@@ -266,6 +270,64 @@ func TestSDKEnd2End(t *testing.T) {
 				grpc.ChainStreamInterceptor(i.StreamInterceptor))
 			pb.RegisterTestServiceServer(s, &testServer{})
 			go s.Serve(lis)
+			defer s.Stop()
+
+			// Establish a connection to the server.
+			clientConn, err := grpc.Dial(lis.Addr().String(), grpc.WithInsecure())
+			if err != nil {
+				t.Fatalf("grpc.Dial(%v) failed: %v", lis.Addr().String(), err)
+			}
+			defer clientConn.Close()
+			client := pb.NewTestServiceClient(clientConn)
+
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			ctx = metadata.NewOutgoingContext(ctx, test.md)
+
+			// Verifying authorization decision for Unary RPC.
+			_, err = client.UnaryCall(ctx, &pb.SimpleRequest{})
+			if got := status.Convert(err); got.Code() != test.wantStatusCode || got.Message() != test.wantErr {
+				t.Fatalf("[UnaryCall] error want:{%v %v} got:{%v %v}", test.wantStatusCode, test.wantErr, got.Code(), got.Message())
+			}
+
+			// Verifying authorization decision for Streaming RPC.
+			stream, err := client.StreamingInputCall(ctx)
+			if err != nil {
+				t.Fatalf("failed StreamingInputCall err: %v", err)
+			}
+			req := &pb.StreamingInputCallRequest{
+				Payload: &pb.Payload{
+					Body: []byte("hi"),
+				},
+			}
+			if err := stream.Send(req); err != nil && err != io.EOF {
+				t.Fatalf("failed stream.Send err: %v", err)
+			}
+			_, err = stream.CloseAndRecv()
+			if got := status.Convert(err); got.Code() != test.wantStatusCode || got.Message() != test.wantErr {
+				t.Fatalf("[StreamingCall] error want:{%v %v} got:{%v %v}", test.wantStatusCode, test.wantErr, got.Code(), got.Message())
+			}
+		})
+	}
+}
+
+func TestSDKFileWatcherEnd2End(t *testing.T) {
+	for name, test := range sdkTests {
+		t.Run(name, func(t *testing.T) {
+			dir := createTmpPolicyFile(t, name+"*", []byte(test.authzPolicy))
```

Comment:
Why are you adding an asterisk here?

Reply:
Removed the asterisk. Earlier, I added that because I thought "*" is replaced with random unique number, but it happens even without the *.